src/main/java/uk/gov/hmcts/juror/api/moj/service/expense/JurorExpenseServiceImpl.java

```diff
@@ -1069,17 +1078,18 @@ static class PendingApprovalMapId {
     @Override
     @Transactional
     @PreAuthorize(SecurityUtil.IS_MANAGER)
-    public void approveExpenses(ApproveExpenseDto dto) {
+    public void approveExpenses(String locCode, PaymentMethod paymentMethod, ApproveExpenseDto dto) {
         ApproveExpenseDto.ApprovalType approvalType = dto.getApprovalType();
-        List<Appearance> appearances = appearanceRepository.findAllByJurorNumberAndPoolNumber(dto.getJurorNumber(),
-                dto.getPoolNumber()).stream()
+        List<Appearance> appearances = appearanceRepository.findAllByCourtLocationLocCodeAndJurorNumber(
+                locCode, dto.getJurorNumber())
+            .stream()
             .filter(appearance -> appearance.getCourtLocation().getOwner().equals(SecurityUtil.getActiveOwner()))
```

Comment:
this looks incorrect based on what we discussed with Liam yesterday - each court location can only approve expenses for their own court location - primary courts should not approve expenses for appearances at satellites of that primary group

Reply:
just this one comment outstanding

Reply:
Updated